code-collab-app/src/editorpage.js

```diff
@@ -6,39 +6,22 @@ import logo from './codeCollabLogo.png';
 import './App.css';
 import IconButton from '@material-ui/core/IconButton';
 import AddIcon from '@material-ui/icons/Add';
+import { inject, observer } from 'mobx-react';
 
 class Editorpage extends React.Component {
   constructor(props){
     super(props);
 
     this.state = {
-      cellCount: 0,
-      cellText: (this.props.location.data) ? this.props.location.data : [],
-      cellLocked: [],
-      cellCurrentlyEditing: null,
+      cellCount: this.props.peer_data.cell_count,
```

Comment:
We need these data on the peer_data side so i just injected peer_data store in this file and using them as prop values. The other way was causing some auto rendering issues, and this seems to be working fine.